SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp

```diff
@@ -606,15 +603,34 @@ void USpatialNetDriver::OnActorSpawned(AActor* Actor)
 	if (!Actor->GetIsReplicated() ||
 		Actor->GetLocalRole() != ROLE_Authority ||
 		!Actor->GetClass()->HasAnySpatialClassFlags(SPATIALCLASS_SpatialType) ||
-		USpatialStatics::IsActorGroupOwnerForActor(Actor))
+		LoadBalanceStrategy == nullptr ||
+		LoadBalanceStrategy->ShouldHaveAuthority(*Actor))
+	{
+		// We only want to delete actors which are replicated and we somehow gain local authority over, when we shouldn't have authority.
+		return;
```

Comment:
Getting authority over something where our lb strategy disagrees seems like an error currently. Once we have dynamic load balancing maybe not but if this is getting hit currently I think we should let ourselves know loudly.

Reply:
The return here is the happy case. We do log an error below if we're forced to delete the actor.